src/brogue/Items.c

```diff
@@ -1629,6 +1617,50 @@ void itemName(item *theItem, char *root, boolean includeDetails, boolean include
     return;
 }
 
+void itemKindName(item *theItem, char *kindName) {
+
+    // use lookup table for randomly generated items with more than one kind per category
+    if (theItem->category & (ARMOR | CHARM | FOOD | POTION | RING | SCROLL | STAFF | WAND | WEAPON)) {
+        strcpy(kindName, tableForItemCategory(theItem->category, NULL)[theItem->kind].name);
+    } else {
+        switch (theItem->category) {
+            case KEY:
+                strcpy(kindName, keyTable[theItem->kind].name); //keys are not randomly generated but have a lookup table
+                break;
+            // these items have only one kind per category and no lookup table
+            case GOLD:
+                strcpy(kindName, "gold peices");
```

Comment:
Yeah, `[--csv] --print-seed-catalog [START NUM LEVELS]` makes sense. I had been thinking about the command line in a hierarchal manner so it didn't occur to me to put the csv switch first. Definitely makes the parsing simpler.

A request on `csvFormat`... I'm a fan of naming boolean variables in a way that indicates such, so how about `isCsvFormat`?

I'm for sure going try for those mega bonus points. Git is less of a magical mystery box now than when I first started using it. Fixups are probably akin to git surgery and I think I'm ready to scrub in. :=)

Reply:
Ok I did my first fixup. Hopefully it qualifies for mega bonus points!

After making the [--csv] change, I gotta say I don't love it. It seems a bit of a non-standard command line usage where a subcommand can only be specified prior to the parent command.

I'm planning on adding more command line options later for corresponding seed catalog features. It might make sense to start looking into a command line parsing library like get-opt.

Reply:
Well I looked into getopt_long yesterday... and when I say "looked into" I mean "refactored 90% of the current code to use getopt_long lol). I don't think it's a good solution for brogue, or really any program that has options with both a long form and short form. Here are the issues:

1. I question whether getopt_long is truly cross-platform. I don't have the knowledge required to fully evaluate this, but I think it only worked on my platform (Windows) because I have mingw installed.

2. Short and long options are defined separately and differently. The definition includes the list of valid options, whether the option has an arg, and whether the arg is required).

3. The feature set differs between short and long options. E.g. there is a nice feature where it will automatically set a flag variable when the option is detected, but it only works on the long form of the option.

4. Argument handling differs. For long options, the argument is expected to be specified with a delimiter on the command line (either a space or "="). For short options, the argument is expected to be part of the same token (i.e. not separated by a space). I will note here that the documentation specifies that a space is also supported for short options but that's not the behavior I saw. It only worked for me if there was no space. The "=" char is not a supported delimiter for short options.